src/relayserver/RelayServer.js

```diff
@@ -609,22 +625,41 @@ class RelayServer extends EventEmitter {
     }
     debug('resending unconfirmed transactions')
     // Get nonce at confirmationsNeeded blocks ago
-    const confirmedBlock = blockHeader.number - confirmationsNeeded
-    debug('signer, blockHeader, confirmedBlock', signer, blockHeader, confirmedBlock)
-    let nonce = await this.web3.eth.getTransactionCount(signer, confirmedBlock)
-    debug('nonce', nonce, confirmedBlock)
-    debug(
-      `resend ${signerIndex}: Removing confirmed txs until nonce ${nonce - 1}. confirmedBlock: ${confirmedBlock}. block number: ${blockHeader.number}`)
+    // const confirmedBlock = Math.max(0, blockHeader.number - confirmationsNeeded)
+    // debug('signer, blockHeader, confirmedBlock', signer, blockHeader, confirmedBlock)
+    // let nonce = await this.web3.eth.getTransactionCount(signer, confirmedBlock)
+    // debug('nonce', nonce, confirmedBlock)
+    // debug(
+    //   `resend ${signerIndex}: Removing confirmed txs until nonce ${nonce - 1}. confirmedBlock: ${confirmedBlock}. block number: ${blockHeader.number}`)
     // Clear out all confirmed transactions (ie txs with nonce less than the account nonce at confirmationsNeeded blocks ago)
-    await this.txStoreManager.removeTxsUntilNonce({ signer, nonce: nonce - 1 })
-
+    // await this.txStoreManager.removeTxsUntilNonce({
+    //   signer,
+    //   nonce: nonce - 1
+    // })
+
+    for (const transaction of sortedTxs) {
+      const receipt = await this.web3.eth.getTransaction(transaction.txId)
```

Comment:
You don't have to fix now, but this is technically the tx itself and not the receipt.

Reply:
Won't fix